<!--
  spwan two crawbots
-->
<launch>
    <param name="/use_sim_time" value="true" />
    <arg name="x1"   default="0"/>
    <arg name="y1"   default="0"/>
    <arg name="z1"   default="0"/>
    <arg name="x2"   default="0"/>
    <arg name="y2"   default="0"/>
    <arg name="z2"   default="0"/>


    <arg name="world_file"     default="$(find crawbot)/worlds/April20/graduate-scenario-1s"/>


    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="$(arg world_file)"/>
    </include>

    <param name="robot_description" command="$(find xacro)/xacro.py $(find crawbot)/models/crawbot.urdf" />

    <!-- BEGIN ROBOT 1-->
    <group ns="crawbot1">
        <param name="tf_prefix" value="/crawbot1_tf" />

        <include file="$(find crawbot)/launch/singlebot.launch" >
            <arg name="init_pose" value="-x $(arg x1) -y $(arg y1) -z $(arg z1)" />
            <arg name="robot_name"  value="crawbot1" />
        </include>

    </group>

    <arg name="yaw" default="-1.57"/>

    <group ns="crawbot2">
        <param name="tf_prefix" value="/crawbot2_tf" />

        <include file="$(find crawbot)/launch/singlebot.launch" >
            <arg name="init_pose" value="-x $(arg x2) -y $(arg y2) -z $(arg z2) -Y $(arg yaw)" />
            <arg name="robot_name"  value="crawbot2" />
        </include>

    </group>


    <rosparam file="$(find crawbot)/config/crawbot_control1.yaml" command="load"/>
    <rosparam file="$(find crawbot)/config/crawbot_control2.yaml" command="load"/>

    <node name="controller_spawner1" pkg="controller_manager" type="spawner" respawn="false"
        output="screen" ns="/crawbot1" args="joint_state_controller
         left_finger_tip_controller  right_finger_tip_controller "/>

    <node name="controller_spawner2" pkg="controller_manager" type="spawner" respawn="false"
         output="screen" ns="/crawbot2" args="joint_state_controller
        left_finger_tip_controller  right_finger_tip_controller "/>


</launch>
